### PR TITLE
Fix an extra var corner case

### PR DESCRIPTION
**Pull request by ForbesLindesay**
See test case update

### CODE REVIEW

**Comment by cpojer** [comment]
seems good!

File: transforms/no-vars.js [hunk 1]
```diff
@@ -24,6 +24,12 @@ export default function(file, api) {
     }
     return {scopeNode, isInFor};
   };
+  const findFunctionDeclaration = (node, container) => {
```
Comment (iamdustan) [inline transforms/no-vars.js]: Why not ArrowFunctionExpressions, Generator functions, or other execution/block scopes?

Comment (ForbesLindesay) [inline transforms/no-vars.js]: They don't get hoisted, so you can't call them early.

Comment (iamdustan) [inline transforms/no-vars.js]: Ah! Now I see the edge case :)
